### ISSUE TITLE
Cant use it on Widget

### ISSUE THREAD

**Issue by RP1-MVac** [Tenpi/soundcloud-extension#10]
Idk if its normal, that it wont work on the soundcloud widget(embedded player) but it doesnt work and it would be very nice if it did

**Comment by Tenpi** [comment]
No, it doesn't work on widgets. Tbh, I don't see much of a point to adding that since it only takes a single click to open the track on SC's website.

**Comment by RP1-MVac** [comment]
but there are widgets were that isn't possible

**Comment by RP1-MVac** [comment]
I found several widgets where, when I click to open SC, the account is empty

**Comment by RP1-MVac** [comment]
so I think some people would be happy if you would add it

**Comment by Tenpi** [comment]
> I found several widgets where, when I click to open SC, the account is empty

Would you mind linking an example of this?

**Comment by RP1-MVac** [comment]
https://harrypotterfanatic.com/audiobook/deutsch/harry-potter-und-die-heiligtumer-des-todes-horbuch-1/ 



it's German but you will see it

**Comment by Tenpi** [comment]
> https://harrypotterfanatic.com/audiobook/deutsch/harry-potter-und-die-heiligtumer-des-todes-horbuch-1/
> 
> it's German but you will see it

Oh, it's because that playlist is set to private. Here is the link to it:
https://soundcloud.com/books-for-soul/sets/hp-7-harry-potter-und-die-heiligtumer-des-todes/s-nNgAx

It's not friendly, but you can find the direct link to these private widgets by inspecting the network tab in your browser's dev tools. Look for any requests to https://api-widget.soundcloud.com/. 

I'm not sure if I will add support for widgets, but you can use the workaround above to get the link and then download it on SC's website.

**Comment by RP1-MVac** [comment]
thank you so much I tried to figure that out for literally a year

**Comment by RP1-MVac** [comment]
but I'm no web developer so I don't have any idea about it

**Comment by RP1-MVac** [comment]
> > https://harrypotterfanatic.com/audiobook/deutsch/harry-potter-und-die-heiligtumer-des-todes-horbuch-1/
> > it's German but you will see it
> 
> Oh, it's because that playlist is set to private. Here is the link to it:
> https://soundcloud.com/books-for-soul/sets/hp-7-harry-potter-und-die-heiligtumer-des-todes/s-nNgAx
> 
> It's not friendly, but you can find the direct link to these private widgets by inspecting the network tab in your browser's dev tools. Look for any requests to https://api-widget.soundcloud.com/.
> 
> I'm not sure if I will add support for widgets, but you can use the workaround above to get the link and then download it on SC's website.

Could you might explain exactly how to to that

**Comment by Tenpi** [comment]
> Could you might explain exactly how to to that

Ok, open the developer tools with Ctrl+Shift+I and navigate to the tab that says "Network".
Refresh the page and you should see a list of all the requests made on the page. Look for the one that was made to https://api-widget.soundcloud.com/.  

For this example, this was the request URL:
https://api-widget.soundcloud.com/playlists/948487585?representation=full&secret_token=s-nNgAx&format=json&client_id=LBCcHmRB8XSStWL6wKH2HPACspQlXg2P&app_version=1626880186

Once you've found it, go to the tab that says "Preview" and you should see the response object. It will look like this:
```js
{
  "artwork_url": "https://i1.sndcdn.com/artworks-000655429267-wx8y9s-large.jpg",
  "created_at": "2019-12-19T14:49:25Z",
  "description": "",
  "duration": 92332514,
  "embeddable_by": "all",
  "genre": "",
  "id": 948487585,
  "kind": "playlist",
  "label_name": null,
  "last_modified": "2019-12-19T14:49:27Z",
  "license": "all-rights-reserved",
  "likes_count": 5,
  "managed_by_feeds": false,
  "permalink": "hp-7-harry-potter-und-die-heiligtumer-des-todes",
  "permalink_url": "https://soundcloud.com/books-for-soul/sets/hp-7-harry-potter-und-die-heiligtumer-des-todes/s-nNgAx"
}
```
The key property here is the one that says "permalink_url", which will be the URL to the playlist from the widget. Let me know if there's anything else that you don't understand.

**Comment by RP1-MVac** [comment]
thank you so much. GitHub is so much friendlier than Reddit
